### PR TITLE
Allow grumphp-shim to 2.x version.

### LINKED ISSUE #25: Upgrade to grumphp v2
As per the changelog, there is no change in usage of GrumPHP. However, the task extension system has changed but this does not affect this repository.

### PR DESCRIPTION
**Pull request by rajeshreeputra**
Fixes #25

### CODE REVIEW

**Comment by hussainweb** [comment]
Thanks, @rajeshreeputra. Is there any reason we should keep support for v1 of the shim? I don't think it has dependencies and so chances of conflicts with other packages is unlikely.

**Comment by rajeshreeputra** [comment]
> Thanks, @rajeshreeputra. Is there any reason we should keep support for v1 of the shim? I don't think it has dependencies and so chances of conflicts with other packages is unlikely.

Agree Updated to use 2.x version. Thank you!!

File: composer.json [hunk 1]
```diff
@@ -27,7 +27,11 @@
         "source": "https://github.com/axelerant/drupal-quality-checker"
     },
     "config": {
-        "sort-packages": true
+        "sort-packages": true,
+        "allow-plugins": {
+            "phpro/grumphp-shim": true,
+            "dealerdirect/phpcodesniffer-composer-installer": true
+        }
```
Comment (hussainweb) [inline composer.json:34]: I am a little concerned with this change and haven't been able to test this. I'll try to do that today. If you can help me understand this better, I can get to it sooner.

My thought is that this probably doesn't help. I am not sure if this configuration takes effect in a package composer.json. There are some settings (such as `repositories`) that can only work in root composer.json. I think this is also one of those settings but I can't find any information about this.

**Review by hussainweb — CHANGES_REQUESTED** [review]
Thanks for your work so far. Please look at my comment.

File: composer.json (hunk 1 above)
Comment (rajeshreeputra) [reply composer.json:34]: This is saying that these plugins are trusted to make changes in project such as on POST_PACKAGE_INSTALL, POST_PACKAGE_UPDATE etc.
This is mainly useful for development, @hussainweb please provide your opinion on should we keep or not.

Comment (hussainweb) [reply composer.json:34]: Ah, you're right. It does not affect other sites using this package but we still need this for our own development.

**Comment by rajeshreeputra** [comment]
@hussainweb when can we expect a release.

**Comment by hussainweb** [comment]
@rajeshreeputra, since this is going to be a breaking release, I will release a new major version. For that, I wanted to change a few more things. Can I get your thoughts on #12 and #21?

**Comment by rajeshreeputra** [comment]
> @rajeshreeputra, since this is going to be a breaking release, I will release a new major version. For that, I wanted to change a few more things. Can I get your thoughts on #12 and #21? I also created #29 for a generic discussion.

I believe a major version isn't necessary in this case; version 1.4.0 should suffice for this. 
We can proceed with maintenance on the 1.3.x version of phpro/grumphp-shim, addressing any issues that may arise in the 1.x version of phpro/grumphp-shim and resolving them accordingly.

**Comment by hussainweb** [comment]
Since we are moving to a new major version of a major dependency (GrumPHP), it is a good idea to make a new major release. I don't know what's changed between GrumPHP 1 and 2 but if there are significant changes, then users have to consciously choose to update.

**Pull request merged**